### PR TITLE
 New package: gnome-latex-3.30.2

### PR DESCRIPTION
**Pull request by Cogitri**
Needs #2454 because of gtksourceview4

### CODE REVIEW

**Comment by Cogitri** [comment]
Hm, it seems like this doesn't work with our current approach of leaving all binaries of texlive-bin in /opt and setting PATH accordingly with `/etc/profile.d/texlive.sh`

File: srcpkgs/perl-latexmk/template [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Template file for 'perl-latexmk'
+pkgname=perl-latexmk
+version=459
+revision=1
+wrksrc="${pkgname/perl-/}"
+build_style=fetch
+hostmakedepends="unzip"
+depends="perl texlive-bin"
+noarch=yes
```
Comment (maxice8) [inline srcpkgs/perl-latexmk/template]: please move noarch=yes to directly below revision=

File: srcpkgs/perl-latexmk/template [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+short_desc="A make-like tool for building LaTeX documents"
+maintainer="Rasmus Thomsen <rasmus.thomsen@protonmail.com>"
+homepage="http://personal.psu.edu/jcc8//software/latexmk/"
```
Comment (maxice8) [inline srcpkgs/perl-latexmk/template]: please move homepage= to directly above distfiles=

File: srcpkgs/tepl/template [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# Template file for 'tepl'
+pkgname=tepl
+version=4.2.0
+revision=1
+build_style=gnu-configure
+configure_args="$(vopt_enable gir introspection)"
+hostmakedepends="glib-devel pkg-config $(vopt_if gir gobject-introspection)"
+makedepends="amtk-devel glib-devel gtksourceview4-devel gtk+3-devel
```
Comment (maxice8) [inline srcpkgs/tepl/template]: shouldn't be libglib-devel ?

**Comment by newbluemoon** [comment]
Is this a problem when building the package or at runtime (or both)?

**Comment by Cogitri** [comment]
> Is this a problem when building the package or at runtime (or both)?

This is only a problem during runtime. I've created an issue for it:

https://github.com/void-linux/void-packages/issues/2543

**Comment by newbluemoon** [comment]
Hmm, I find it a bit strange that gnome-latex cannot be made to find it in /opt. Cannot test-build right now because of gtk+3.

However, latexmk is part of texlive (`tlmgr info latexmk`).

**Comment by Cogitri** [comment]
> However, latexmk is part of texlive (tlmgr info latexmk).

Yes, but it isn't installed by default. Prompting the user to install it with `tlmgr` before using `gnome-latex` didn't seem like an optimal solution to me.

> Hmm, I find it a bit strange that gnome-latex cannot be made to find it in /opt.

Well, it is possible to change the `latexmk` arguments, but this means that you have to dig into the settings and change the command for each of the different build methods `gnome-latex` offers one by one (which is really annoying and not user friendly)

**Comment by newbluemoon** [comment]
> Prompting the user to install it with tlmgr before using gnome-latex didn't seem like an optimal solution to me.

But it wouldn’t be a precedence. ;) See e.g. `lyx/INSTALL.msg`.

**Comment by Cogitri** [comment]
> But it wouldn’t be a precedence. ;) See e.g. lyx/INSTALL.msg.

Oh, yeah, forgot about INSTALL hooks, I'm still a little new to Void, I guess :P 
Thanks for the heads up

**Comment by newbluemoon** [comment]
Just tested: when you use latexmk from texlive it works right out of the box!

Maybe add an `INSTALL.msg` like, e.g.:

> gnome-latex uses latexmk by default. It is recommended
> to add it to your texlive installation:
>
>tlmgr install latexmk

**Comment by Cogitri** [comment]
> Just tested: when you use latexmk from texlive it works right out of the box!

Oh, neat, thanks for testing that!

**Comment by newbluemoon** [comment]
You’re welcome. :)

**Comment by Cogitri** [comment]
Hm, does it autodetect latexmk for you though?

**Comment by newbluemoon** [comment]
Yes, I just installed latexmk with tlmgr and gnome-latex, opened a .tex file and hit the LaTeX --> PDF icon.
If it doesn’t work for you it’s probably because you have your own perl-latexmk installed?
`/etc/profile.d/texlive.sh` appends to `PATH` so you’re version comes first. Just remove it.

**Comment by Cogitri** [comment]
I uninstalled my latexmk first and installed latexmk via tlmgr, but gnome-latex only returns `Failed to execute child process "latexmk" (No such file or directory)`

**Comment by newbluemoon** [comment]
And when you just type it on the command line? Is your `PATH` set up correctly?

**Comment by Cogitri** [comment]
No, seems like PATH isn't correct. I use ZSH as shell (it's run via gnome-terminal), but I didn't change my login shell (so I guess it's still DASH)

**Comment by newbluemoon** [comment]
`man dash`
>     ... A login
>     shell first reads commands from the files /etc/profile and .profile if
>     they exist.  If the environment variable ENV is set on entry to an
>     interactive shell, or is set in the .profile of a login shell, the shell
>     next reads commands from the file named in ENV.  Therefore, a user should
>     place commands that are to be executed only at login time in the .profile
>     file, and commands that are executed for every interactive shell inside
>     the ENV file. ...

But you could try `chsh` to zsh.

**Comment by Cogitri** [comment]
Okay, figured it out, seems like rustup manipulated the path somehow

**Comment by newbluemoon** [comment]
#2543 should be resolved then? ;)

**Comment by maxice8** [comment]
Can it be merged after common/shlibs conflicts are solved ?

**Comment by Cogitri** [comment]
Yup

**Pull request merged**